test/simulator/items/whiteherb.js

```diff
@@ -0,0 +1,150 @@
+			{species: "pelipper", ability: 'sandveil', item: 'lifeorb', moves: ['protect']}]);
+
+		battle.makeChoices('move growl 0, move growl 0', 'move autotomize, move protect');
+
+		const holder = battle.p2.active[0];
+		assert.false.holdsItem(holder);
+		assert.statStage(holder, 'atk', -1);
+	});
+	it('should use white herb after partingshot during active turn', function () {
+		battle = common.createBattle({gameType: 'doubles'});
+
+		battle.join('p1', 'Guest 1', 1, [
+			{species: "chansey", ability: 'shellarmor', item: 'ironball', moves: ['partingshot']},
+			{species: "clefable", ability: 'levitate', item: 'ironball', moves: ['partingshot']},
+			{species: "aerodactyl", ability: 'shellarmor', item: 'ironball', moves: ['stealthrock']},
+			{species: "rotom", ability: 'levitate', item: 'ironball', moves: ['rest']}]);
+		battle.join('p2', 'Guest 2', 2, [
+			{species: "aegislash", ability: 'stancechange', item: 'whiteherb', moves: ['autotomize']},
+			{species: "pelipper", ability: 'sandveil', item: 'lifeorb', moves: ['protect']}]);
+
+		battle.makeChoices('move partingshot 0, move partingshot 0', 'move autotomize, move protect');
+		battle.makeChoices('switch aerodactyl, switch rotom', 'move autotomize, move protect');
+
+		const holder = battle.p2.active[0];
+		assert.false.holdsItem(holder);
+		assert.statStage(holder, 'atk', -1);
+	});
+	it('should use white herb after both intimidate', function () {
+		let battle;
+		battle = common.createBattle({gameType: 'doubles'});
+
+		battle.join('p1', 'Guest 1', 1, [{species: "Arcanine", ability: 'intimidate', moves: ['bodyslam']},
+			{species: "Incineroar", ability: 'intimidate', moves: ['agility']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: "aegislash", ability: 'stancechange', item: 'whiteherb', moves: ['gyroball']},
+			{species: "pelipper", ability: 'sandveil', item: 'lifeorb', moves: ['airslash']}]);
+
+
+		battle.makeChoices('move bodyslam, move agility', 'move gyroball, move airslash');
+
+		const holder = battle.p2.active[0];
+
+		assert.false.holdsItem(holder);
+		assert.statStage(holder, 'atk', 0);
+	});
+	it('should use white herb after one intimidate', function () {
+		let battle;
+		battle = common.createBattle({gameType: 'doubles'});
+
+		battle.join('p1', 'Guest 1', 1, [{species: "Arcanine", ability: 'flashfire', moves: ['bodyslam']},
+			{species: "Incineroar", ability: 'intimidate', moves: ['agility']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: "aegislash", ability: 'stancechange', item: 'whiteherb', moves: ['gyroball']},
+			{species: "pelipper", ability: 'sandveil', item: 'lifeorb', moves: ['airslash']}]);
+
+
+		battle.makeChoices('move bodyslam, move agility', 'move gyroball, move airslash');
+
+		const holder = battle.p2.active[0];
+
+		assert.false.holdsItem(holder);
+		assert.statStage(holder, 'atk', 0);
+	});
+	it('should use white herb after two intimidate switch in', function () {
```

Comment:
Switching in during a turn (one at a time) is fundamentally different from switching in at the beginning of a battle or to replace fainted Pokemon (all at once). This test you're doing here should activate White Herb before the second Pokemon even switches out.

Reply:
ah ok makes sense, will update the PR

Reply:
> Switching in during a turn (one at a time) is fundamentally different from switching in at the beginning of a battle or to replace fainted Pokemon (all at once). This test you're doing here should activate White Herb before the second Pokemon even switches out.

@Marty-D 

It seems runSwitch remains in queue when Pokemon are switching in at start or after feint. Checking for this before the update leads to White Herb being called after all Pokemon switch in after start or after feint and calls it right away after a player switches during a turn. Is this correct behavior?

Reply:
```suggestion
	it('should use white herb after two Intimidate switch in', function () {
```